### PR TITLE
Rename order status column

### PR DESCRIPTION
**Pull request by islean**
## Description

Part of https://github.com/Clinical-Genomics/cigrid-ui/issues/638. Renames the `is_delivered` column in `Order` to `is_open` and flips all the values. Needs to be merged with the FE changes in https://github.com/Clinical-Genomics/cigrid-ui/pull/691.

### Added

-

### Changed

- The is_delivered column in Order is renamed to is_open with values flipped.

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b rename-order-status-column -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by islean** [comment]
Works on stage but need to check with the FE:

image

**Comment by islean** [comment]
Works with the FE:

image

File: cg/services/orders/order_service/order_service.py [hunk 1]
```diff
@@ -41,15 +41,15 @@ def create_order(self, order_data: OrderIn) -> OrderResponse:
...
-    def set_delivery(self, order_id: int, delivered: bool) -> OrderResponse:
-        order: Order = self.store.update_order_delivery(order_id=order_id, delivered=delivered)
+    def set_open(self, order_id: int, open: bool) -> OrderResponse:
+        order: Order = self.store.update_order_delivery(order_id=order_id, open=open)
         return create_order_response(order)
 
-    def update_delivered(self, order_id: int, delivered_analyses: int) -> None:
-        """Update the delivery status of an order based on the number of delivered analyses."""
+    def update_is_open(self, order_id: int, delivered_analyses: int) -> None:
+        """Update the openness status of an order based on the number of delivered analyses."""
```
Comment (diitaz93) [inline cg/services/orders/order_service/order_service.py:49]:
```suggestion
        """Update the is_open attribute of an order based on the number of delivered analyses."""
```

File: cg/server/endpoints/orders.py [hunk 2]
```diff
@@ -73,24 +72,24 @@ def get_order(order_id: int):
...
         return jsonify(response_data.model_dump()), HTTPStatus.OK
     except OrderNotFoundError as error:
         return jsonify(error=str(error)), HTTPStatus.NOT_FOUND
 
 
-@ORDERS_BLUEPRINT.route("/orders/<order_id>/update-delivery-status", methods=["POST"])
+@ORDERS_BLUEPRINT.route("/orders/<order_id>/update-open-status", methods=["POST"])
 def update_order_delivered(order_id: int):
-    """Update the delivery status of an order based on the number of delivered analyses."""
+    """Update the openness status of an order based on the number of delivered analyses."""
```
Comment (diitaz93) [inline cg/server/endpoints/orders.py:88]:
```suggestion
    """Update the is_open attribute of an order based on the number of delivered analyses."""
```

File: cg/server/endpoints/orders.py [hunk 3]
```diff
@@ -73,24 +72,24 @@ def get_order(order_id: int):
...
+        is_open: bool = request_data.is_open
+        response_data: Order = order_service.set_open(order_id=order_id, open=is_open)
         return jsonify(response_data.model_dump()), HTTPStatus.OK
     except OrderNotFoundError as error:
         return jsonify(error=str(error)), HTTPStatus.NOT_FOUND
 
 
-@ORDERS_BLUEPRINT.route("/orders/<order_id>/update-delivery-status", methods=["POST"])
+@ORDERS_BLUEPRINT.route("/orders/<order_id>/update-open-status", methods=["POST"])
 def update_order_delivered(order_id: int):
```
Comment (diitaz93) [inline cg/server/endpoints/orders.py:87]: can this function name be changed to reflect the changes in the attribute?

File: cg/server/endpoints/orders.py [hunk 4]
```diff
@@ -73,24 +72,24 @@ def get_order(order_id: int):
         return make_response(jsonify(error=str(error)), HTTPStatus.NOT_FOUND)
 
 
-@ORDERS_BLUEPRINT.route("/orders/<order_id>/delivered", methods=["PATCH"])
+@ORDERS_BLUEPRINT.route("/orders/<order_id>/open", methods=["PATCH"])
 def set_order_delivered(order_id: int):
```
Comment (diitaz93) [inline cg/server/endpoints/orders.py:76]: If possible, update the name of this function

**Review by diitaz93 — APPROVED** [review]
Looks great! 💯

File: alembic/versions/2024_08_09_7770dcad8bde_rename_order_status.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+import sqlalchemy as sa
+from sqlalchemy import orm
+from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "7770dcad8bde"
+down_revision = "bb4c6dbad991"
```
Comment (diitaz93) [inline alembic/versions/2024_08_09_7770dcad8bde_rename_order_status.py:15]: update down migration :)

File: cg/server/endpoints/orders.py (hunk 4 above)
Comment (islean) [reply cg/server/endpoints/orders.py]: Updated

File: alembic/versions/2024_08_09_7770dcad8bde_rename_order_status.py (hunk 5 above)
Comment (islean) [reply alembic/versions/2024_08_09_7770dcad8bde_rename_order_status.py:15]: In your PR ;)

**Comment by islean** [comment]
Deployed to hasta and the VMs:

```
Log deploy... done.
cg, version 63.1.0
remote: Enumerating objects: 14, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (9/9), done.
remote: Total 14 (delta 6), reused 9 (delta 4), pack-reused 0 (from 0)
Unpacking objects: 100% (14/14), done.
From https://github.com/Clinical-Genomics/cg
   9dbfb30..e957bea  master     -> origin/master
   5ca5457..0521273  patch_taxprofiler_versions -> origin/patch_taxprofiler_versions
 * [new tag]         v63.1.0    -> v63.1.0
Already on 'master'
Your branch is behind 'origin/master' by 2 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
Updating 9dbfb30..e957bea
Fast-forward
 .bumpversion.cfg                                                |  2 +-
 alembic/versions/2024_08_09_7770dcad8bde_rename_order_status.py | 50 ++++++++++++++++++++++++++++++++++++++++++++++++++
 cg/__init__.py                                                  |  2 +-
 cg/server/admin.py                                              |  2 +-
 cg/server/dto/orders/order_delivery_update_request.py           |  2 +-
 cg/server/dto/orders/order_patch_request.py                     |  4 ++--
 cg/server/dto/orders/orders_request.py                          |  2 +-
 cg/server/dto/orders/orders_response.py                         |  2 +-
 cg/server/endpoints/orders.py                                   | 27 ++++++++++++---------------
 cg/services/orders/order_service/order_service.py               | 18 +++++++++---------
 cg/services/orders/order_service/utils.py                       |  4 ++--
 cg/store/crud/read.py                                           |  4 ++--
 cg/store/crud/update.py                                         |  6 +++---
 cg/store/filters/status_order_filters.py                        | 10 +++++-----
 cg/store/models.py                                              |  2 +-
 pyproject.toml                                                  |  2 +-
 16 files changed, 93 insertions(+), 46 deletions(-)
 create mode 100644 alembic/versions/2024_08_09_7770dcad8bde_rename_order_status.py
/home/isak.ohlsson
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade bb4c6dbad991 -> 7770dcad8bde, Rename order status
Revision ID: 7770dcad8bde
Revises: bb4c6dbad991
Create Date: 2024-08-09 09:47:47.814700
[isak.ohlsson@hasta:~] [P_base] $ 
```

**Pull request merged**